### PR TITLE
Sydney rating form

### PR DESCRIPTION
**Pull request by XhenryZhang**
Features:
* adding/editing/deleting review (one review per user per item)
* display average rating of item
* form validations such that score is between 0 and 5, inclusive

### CODE REVIEW

File: app/javascript/components/RatingForm/RatingForm.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { createRating, updateRating, deleteRating } from '../../actions/ratings';
+
+const RatingForm = ( { itemId, item } ) => {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+    const location = useLocation();
+    const currState = useSelector((state) => state);
+    const ratings = useSelector((state) => state.ratings);
+
+    console.log(currState);
```
Comment (XuperBryan) [inline app/javascript/components/RatingForm/RatingForm.js:17]: small edit: remove console.logs if not neccessary

File: app/javascript/components/RatingForm/RatingForm.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        setUser(JSON.parse(localStorage.getItem('profile')));
+    }, [location]);
+
+    const clear = () => {
+        setRatingData({score: '', comment: '', validScore: false});
+    }
+
+    const validateRating = (rating) => {
+        const ratingInt = parseInt(rating);
+        return ratingInt >= 0 && ratingInt <= 5;
```
Comment (XuperBryan) [inline app/javascript/components/RatingForm/RatingForm.js:77]: are we allowing 0 stars? not sure if we are, just wanted to check :O

**Review by XuperBryan — APPROVED** [review]
One comment! but overall LGTM

File: app/javascript/components/RatingForm/RatingForm.js (hunk 2 above)
Comment (XuperBryan) [reply app/javascript/components/RatingForm/RatingForm.js:77]: just saw the features description, nevermind :D

**Comment by just-in-chang** [comment]
Helped rebase ^

**Pull request merged**